lib/commands/touch.js

```diff
@@ -1,9 +1,99 @@
+import { util } from 'appium/support';
+
 const commands = {};
+const { promisify } = require('util');
+const sleep = promisify(setTimeout);
 
 //This is needed to make clicks on -image elements work properly
 commands.performActions = async function (actions) {
-  return await this.winAppDriver.sendCommand('/actions', 'POST', {actions});
+  //Workaround for https://github.com/appium/appium/issues/16268
+  //Once WAD offers full W3C support for keyboard and mouse type pointer actions, this can be proxied directly to the /actions endpoint
+  for (const command in actions) {
```

Comment:
just to confirm this approach is the expected one. Basically all items in this array represent actions that must be executed in parallel (e.g. keys and mouse gestures represented by different items must be executed in parallel unless ticks don't match), but here we run them in the serial order.

Reply:
Here we do run them in serial order. 
It's not clear to me when would they be executed in parallel?
Because if I execute chained commands then they will executed one by one (ex: actions.moveTO(element).click().perform()
The mouse moves then it does the click action.

Also, if the same command contains multiple steps. 
actions.dragAndDrop(from, to).perform() Behind the scenes the command represents basically a moveMouse(from) -> pointerDown -> moveMouse(to) -> pointerUp
So this kind of command that contains chained actions is also done in a serial way

I also haven't encountered any Actions command who require a parallel execution command. But I might be missing something?

Reply:
ok, simple question, - how to implement "zoom" gesture using actions?